cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java

```diff
@@ -168,37 +282,89 @@ public void testBuildCollectionNameDMByServicePathRootServicePath() {
                         + "- FAIL - There was some problem when building the collection name");
             assertTrue(false);
         } // catch
-    } // testBuildCollectionNameDMByServicePathRootServicePath
+    } // testBuildCollectionNameDMByServicePathRootServicePathNewEncoding
+    
+    /**
+     * [NGSIMongoBaseSink.buildCollectionName] -------- When / service-path is notified/defaulted and
+     * data_model=dm-by-entity, the MongoDB collections name is the concatenation of the \<prefix\>, \<service-path\>,
+     * \<entityId\> and \<entityType\>.
+     */
+    @Test
+    public void testBuildCollectionNameDMByEntityRootServicePathOldEncoding() {
+        System.out.println(getTestTraceHead("[NGSIMongoBaseSink.buildCollectionName]")
+                + "-------- When / service-path is notified/defaulted and data_model=dm-by-entity, the MongoDB"
+                + "collections name is the concatenation of the <prefix>, <service-path>, <entityId> and <entityType>");
+        String collectionPrefix = "sth_";
+        String dbPrefix = "sth_";
+        String dataModel = "dm-by-entity";
+        String enableEncoding = "false";
+        NGSIMongoBaseSinkImpl sink = new NGSIMongoBaseSinkImpl();
+        sink.configure(CommonUtilsForTests.createContextForMongoSTH(collectionPrefix, dbPrefix, dataModel,
+                enableEncoding));
+        String dbName = "sth_default";
+        String fiwareService = "default";
+        String fiwareServicePath = "/";
+        String entity = "someId=someType";
+        String attribute = "someName";
+        boolean isAggregated = false;
+        String entityId = "someId";
+        String entityType = "someType";
+        
+        try {
+            String collectionName = sink.buildCollectionName(dbName, fiwareServicePath, entity, attribute,
+                    isAggregated, entityId, entityType, fiwareService);
+            String expectedCollectionName = "sth_/_someId_someType";
+            
+            try {
+                assertEquals(expectedCollectionName, collectionName);
+                System.out.println(getTestTraceHead("[NGSIMongoBaseSink.buildCollectionName]")
+                        + "-  OK  - '" + collectionName + "' was crated as collection name");
+            } catch (AssertionError e) {
+                System.out.println(getTestTraceHead("[NGSIMongoBaseSink.buildCollectionName]")
+                        + "- FAIL - '" + collectionName + "' was crated as collection name instead of "
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4